File: CSharp/Strategy.Duck/DecoyDuck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPatterns.Strategy
{
    public class DecoyDuck : Duck
    {
        public DecoyDuck()
        {
            flyBehaviour = new FlyNoWay();
            quackBehaviour = new MuteQuack();
        }

        public override void display()
        {
            Console.Out.WriteLine("I am a Decoy Duck");
        }
    }
}

File: Java/hfdp/src/main/java/chasej/hfdp/strategy/DecoyDuck.java
package chasej.hfdp.strategy;

public class DecoyDuck extends Duck {
	//Constructor
	public DecoyDuck() {
		setFlyBehavior(new FlyNoWay());
		setQuackBehavior(new MuteQuack());
	}

	public void display() {
		System.out.println("I'm a duck decoy");
	}

}

